Community Board Update

zachseidner1 opened the pull request:
## Overview

This update adds the the community board for Android using Vin's backend, including the trending screen, flyers screen, and reads screen. 


## Changes Made

- Whole app restructured, Trending is now home screen, Reads page contains articles, magazines, and publications. Flyers screen shows community board. 
- Articles now have shimmer loading effect in new screens. 
- Networking for Flyers not yet implemented, but everything else works.


## Next Steps

- Integrate actual backend once it comes out.


## Screenshots & Videos
See [here](https://drive.google.com/drive/folders/1HXlNHt8BIN2DdRWNhY0EoW84D-p3jPV4?usp=share_link)

zachseidner1 committed:
Implemented Article Shuffling
- The articles under Other Articles are now shuffled by publication
- If the shuffled articles failed it should default to the previous query

zachseidner1 committed:
Basic layout for Reads page finished
- Still need to add funcitonality to the tabs switcher
- Still need to add the publications viewer with the hamburger selector
- Need to add this as a screen but the initial layout is done

zachseidner1 committed:
Reads screen complete
- Reads screen finished for community board update
- Still need to update home screen to show trending
- Still need to implement flyers page
- Everything for the reads screen should be working

zachseidner1 committed:
Flyers Page UI Complete
- Updated home and navigation to allow for Flyers page
- Created components for small and big flyers
- Implemented components in scrollable screen

zachseidner1 committed:
Fixed Spacing on Flyers Screen

zachseidner1 committed:
Added shimmer effect for loading flyers

zachseidner1 committed:
Implemented Trending Screen
- Created main featured article component with custom gradient background
- Implemented trending screen with networking (except for Flyers)
- Fixed issues with magazine back button

zachseidner1 committed:
Delete unnecessary routes, remove logs

zachseidner1 committed:
Implement basic framework for networks
- Create relevant models
- Parametrize components

Corfish123 commented on app/src/main/java/com/cornellappdev/android/volume/ui/components/general/FlyerComponents.kt:
```diff
@@ -0,0 +1,191 @@
+import com.cornellappdev.android.volume.ui.theme.VolumeOrange
+import com.cornellappdev.android.volume.ui.theme.lato
+import com.cornellappdev.android.volume.ui.theme.notoserif
+
```
Add a comment here explaining what this composable does

Corfish123 replied:
address these please

Corfish123 commented on app/src/main/java/com/cornellappdev/android/volume/ui/components/general/FlyerComponents.kt:
```diff
@@ -0,0 +1,191 @@
+        IconTextRow(text = "Bailey Hall", iconId = R.drawable.ic_location_pin)
+    }
+}
+
```
same thing here

Corfish123 replied:
add comments

Corfish123 commented on app/src/main/java/com/cornellappdev/android/volume/ui/components/general/FlyerComponents.kt:
```diff
@@ -0,0 +1,191 @@
+        Column (modifier = Modifier.padding(start = 8.dp)) {
+            OrganizationAndIconsRow(organizationName = "Break Free", iconSize = 20.dp)
+            Text(
+                text = "New Destinations",
```
Why are there a bunch of default values here with the location and date, etc?

Corfish123 commented on app/src/main/java/com/cornellappdev/android/volume/ui/components/general/FlyerComponents.kt:
```diff
@@ -0,0 +1,191 @@
+        }
+    }
+}
+
```
Please go through and comment all of your composables

Corfish123 commented on app/src/main/java/com/cornellappdev/android/volume/ui/components/general/TrendingPageComponents.kt:
```diff
@@ -0,0 +1,222 @@
+                }
+            }
+        }
+        Log.d(TAG, "MainArticleComponent: Image bitmap after launched effect: $imageBitmap")
```
delete this

Corfish123 commented on app/src/main/java/com/cornellappdev/android/volume/ui/components/general/TrendingPageComponents.kt:
```diff
@@ -0,0 +1,222 @@
+
+
+private suspend fun getBitmap(imageUrl: String, context: Context): Bitmap? {
+    Log.d(TAG, "getBitmap: Get bitmapped called")
```
delete all of the log statements

Corfish123 commented:
Great work!!! 
A few issues to point out: in the flyer tab, in the upcoming portion, the location is slightly cut off. Please fix that.  Other than that wonderful work

zachseidner1 committed:
Continue framework for networks (WIP Commit)

zachseidner1 committed:
Implemented Fake Flyers Repository and Flyers View Model
- Flyers repository is a placeholder until backend does their thing
- Flyers view model should work theoretically when the repository is implemented

zachseidner1 committed:
Nearly finished CBoard with Vin's backend
- Just need to put Flyers on trending page

zachseidner1 committed:
Finish Vin's Backend Implementation
- Community board for Android is fully complete until Vin's backend goes down
- Added sharing and click functionality for flyers
- Added flyers on the trending page

zachseidner1 committed:
Fix other articles infinite load

zachseidner1 committed:
Update to use daily endpoint instead of filtering on frontend

zachseidner1 committed:
Update text for when there are no flyers

zachseidner1 committed:
Fix annoying spacing issue

Corfish123 reviewed (approved):
Great work!!!
